pkg/ctl/utils/update_cluster_endpoint_access.go

```diff
@@ -0,0 +1,168 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+var (
+	private bool
+	public  bool
+)
+
+func updateClusterEndpointsCmd(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("update-cluster-endpoints", "Update Kubernetes API endpoint access configuration", "")
+
+	cmd.SetRunFunc(func() error {
+		return doUpdateClusterEndpoints(cmd, private, public)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		cmdutils.AddNameFlag(fs, cfg.Metadata)
+		cmdutils.AddRegionFlag(fs, cmd.ProviderConfig)
+		cmdutils.AddConfigFileFlag(fs, &cmd.ClusterConfigFile)
+		cmdutils.AddApproveFlag(fs, cmd)
+		cmdutils.AddTimeoutFlag(fs, &cmd.ProviderConfig.WaitTimeout)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("Update private/public Kubernetes API endpoint access configuration",
+		func(fs *pflag.FlagSet) {
+			fs.BoolVar(&private, "private-access", false, "access for private (VPC) clients")
+			fs.BoolVar(&public, "public-access", false, "access for public clients")
+		})
+	cmdutils.AddCommonFlagsForAWS(cmd.FlagSetGroup, cmd.ProviderConfig, false)
+}
+
+func accessFlagsSet(cmd *cmdutils.Cmd) (privateSet, publicSet bool) {
+	cmd.FlagSetGroup.InFlagSet("Update private/public Kubernetes API endpoint access configuration",
+		func(fs *pflag.FlagSet) {
+			fs.VisitAll(func(f *pflag.Flag) {
```

Comment:
can use `fs.Lookup(name)` instead

Reply:
Possibly, though there are only two flags in this flagset anyway, and the difference amounts to one line of code once you account for the lookup possibly returning nil.

Before:
```
		func(fs *pflag.FlagSet) {
			fs.VisitAll(func(f *pflag.Flag) {
				switch f.Name {
				case "private-access":
					privateSet = f.Changed
				case "public-access":
					publicSet = f.Changed
				default:
					// do nothing
				}
			})
```
After:
```
		func(fs *pflag.FlagSet) {
			priv := fs.Lookup("private-access")
			if priv != nil {
				privateSet = priv.Changed
			}
			pub := fs.Lookup("public-access")
			if pub != nil {
				publicSet = pub.Changed
			}
		}
```

Reply:
what about
```
                func(fs *pflag.FlagSet) {
			if priv := fs.Lookup("private-access"); priv != nil {
				privateSet = priv.Changed
			}
			if pub := fs.Lookup("public-access"); pub != nil {
				publicSet = pub.Changed
			}
		}
```
and technically, there is a bug if the lookup returns `nil`. in that case it will just ignore the value passed in.